### ISSUE TITLE
Explicitly setting device orientation in iOS 16 emits a warning: Attempting to set an orientation on UIDevice via setValue:forKey: isn’t supported and no longer works.

### ISSUE THREAD

**Issue by soarescaique** [flutter/flutter#111587]
## Steps to Reproduce

According to iOS16 Release note:

**"Attempting to set an orientation on UIDevice via setValue:forKey: isn’t supported and no longer works."**

After update my system to iOS16 ( Version 16.0 ) if you try to change the screen orientation, you will get the following warning. 

**Expected results:** 

The screen orientation changes without any warnings.

**Actual results:** 

> **BUG IN CLIENT OF UIKIT: Setting UIDevice.orientation is not supported. Please use UIWindowScene.requestGeometryUpdate(_:)**

**Code example:** 

```
 @override
  void initState() {
    SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft]); // This line reproduces the bug

    super.initState();
  }
```

**Flutter Doctor:**

> [✓] Flutter (Channel stable, 3.3.2, on macOS 12.6 ---- darwin-arm, locale en-BR)
> [✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
> [✓] Xcode - develop for iOS and macOS (Xcode 14.0)
> [✓] Chrome - develop for the web
> [✓] Android Studio (version 2021.1)
> [✓] VS Code (version 1.71.1)
> [✓] Connected device (3 available)
> [✓] HTTP Host Availability

**Comment by danagbemava-nc** [comment]
Issue is reproducible on `stable` and `master`.

During testing, I was not able to reproduce this on the iOS Simulator but it did reproduce on my physical iOS device. 

Labeling for further investigation. 

<details>
<summary>sample from logs</summary>

```
[+4360 ms] [Orientation] BUG IN CLIENT OF UIKIT: Setting UIDevice.orientation is not supported. Please use
UIWindowScene.requestGeometryUpdate(_:)
[+2913 ms] [Orientation] BUG IN CLIENT OF UIKIT: Setting UIDevice.orientation is not supported. Please use
UIWindowScene.requestGeometryUpdate(_:)
[+1471 ms] [Orientation] BUG IN CLIENT OF UIKIT: Setting UIDevice.orientation is not supported. Please use
UIWindowScene.requestGeometryUpdate(_:)
```
</details>

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  DeviceOrientation orientation = DeviceOrientation.landscapeLeft;

  @override
  void initState() {
    super.initState();

    SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft]); 
  }

  void _incrementCounter() {
    setState(() {
      _counter++;
      if(orientation == DeviceOrientation.landscapeLeft) {
        orientation = DeviceOrientation.portraitUp;
      }else {
        orientation = DeviceOrientation.landscapeLeft;
      }
      setOrientation(orientation);
    });
  }

  void setOrientation(DeviceOrientation orientation) {
    SystemChrome.setPreferredOrientations([orientation]);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}
```
</details>

[verbose.log](https://github.com/flutter/flutter/files/9573182/verbose.log)

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.3.2, on macOS 12.6 21G115 darwin-arm, locale en-GB)
    • Flutter version 3.3.2 on channel stable at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e3c29ec00c (18 hours ago), 2022-09-14 08:46:55 -0500
    • Engine revision a4ff2c53d8
    • Dart version 2.18.1
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8815526/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A309
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8815526/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.71.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (4 available)
    • Nexus (mobile)         • 00008020-001875E83A38002E            • ios            • iOS 16.0 20A362
    • iPhone 14 Pro (mobile) • 4302007A-B132-4B89-887A-C5993FCA3284 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 3.4.0-19.0.pre.272, on macOS 12.6 21G115 darwin-arm64, locale en-GB)
    • Flutter version 3.4.0-19.0.pre.272 on channel master at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2262892e3b (4 hours ago), 2022-09-15 00:44:22 -0400
    • Engine revision 63cc5ea2b1
    • Dart version 2.19.0 (build 2.19.0-201.0.dev)
    • DevTools version 2.17.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8815526/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A309
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2021.2)
    • Android Studio at /Users/nexus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/212.5712.43.2112.8815526/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.71.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (4 available)
    • Nexus (mobile)         • 00008020-001875E83A38002E            • ios            • iOS 16.0 20A362
    • iPhone 14 Pro (mobile) • 4302007A-B132-4B89-887A-C5993FCA3284 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by jmagman** [comment]
Does the orientation actually change, but there's just a warning? Or is orientation change broken?

**Comment by jmagman** [comment]
I reproduced this, the orientation did change, but there was a warning.
```
2022-09-15 10:14:18.504927-0700 Runner[978:362624] [Orientation] BUG IN CLIENT OF UIKIT: Setting UIDevice.orientation is not supported. Please use UIWindowScene.requestGeometryUpdate(_:)
```

**Comment by soarescaique** [comment]
@danagbemava-nc I also could not reproduce this "Warning" on iOS Simulator. But it does happen on the physical device every time you change the orientation. 

@jmagman The Screen orientation does change, it's a warning. But According to the Apple Release Notes of iOS16 (UIKit Section) it might not be supported in the future, as we now need to use `UIWindowScene.requestGeometryUpdate(_:)` to change the orientation

**Comment by jmagman** [comment]
`-attemptRotationToDeviceOrientation` has also been deprecated, see `-setNeedsUpdateOfSupportedInterfaceOrientations`

https://github.com/flutter/engine/blob/9eb565bd4067225369d1aa2a48b4fef4de93ff53/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm#L1539

**Comment by ghermana** [comment]
Calling
`SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]);`
also causes the two rotation animations: a 90 degree counterclockwise rotation, followed by a 180 degree clockwise rotation. The device ends up rotating in the selected orientation, but the double rotation animation is confusing.

This happens after having called
`SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp, DeviceOrientation.portraitDown, DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]);`
first.

**Comment by GyeomSik** [comment]
Is there any progress??

**Comment by soarescaique** [comment]
@GyeomSik Not as far as I know. I tried to fix it, but did not have success.

**Comment by micodev** [comment]
same issue here please hot fix this :(

**Comment by jmagman** [comment]
> also causes two rotation animations: a 90 degree counterclockwise rotation, followed by a 180 degree clockwise rotation. The device ends up rotating in the selected orientation, but the double rotation animation is confusing.

I was not able to reproduce this, and even if I did it doesn't sound related to this deprecation.
```dart
      if (_counter % 2 == 0) {
        SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp, DeviceOrientation.portraitDown, DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]);
      } else {
        SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]);
      }
```
on a physical phone this rotated between portrait up and landscape left with no issues.

@ghermana can you file a new issue with more complete reproduction steps?

I am only seeing the `[Orientation] BUG IN CLIENT OF UIKIT:` warning, but the orientation is correctly changing.  It needs to be fixed, but the urgency of some of these comments makes me think something other than a stray warning is happening?

**Comment by jmagman** [comment]
> this crash our app when using orientation.

@micodev please file a new issue with the details about the crash so we can investigate further.  I wasn't able to reproduce anything but some logging.

**Comment by kamilponiewierski** [comment]
> > also causes two rotation animations: a 90 degree counterclockwise rotation, followed by a 180 degree clockwise rotation. The device ends up rotating in the selected orientation, but the double rotation animation is confusing.
> 
> I was not able to reproduce this, and even if I did it doesn't sound related to this deprecation.
> 
> ```dart
>       if (_counter % 2 == 0) {
>         SystemChrome.setPreferredOrientations([DeviceOrientation.portraitUp, DeviceOrientation.portraitDown, DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]);
>       } else {
>         SystemChrome.setPreferredOrientations([DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]);
>       }
> ```
> 
> on a physical phone this rotated between portrait up and landscape left with no issues.
> 
> @ghermana can you file a new issue with more complete reproduction steps?
> 
> I am only seeing the `[Orientation] BUG IN CLIENT OF UIKIT:` warning, but the orientation is correctly changing. It needs to be fixed, but the urgency of some of these comments makes me think something other than a stray warning is happening?

Have you attempted to reproduce it in release mode? For us, it hasn't been observed in debug mode, we stil could see the warning log though.

For what it's worth, the issue happened while rotating between `DeviceOrientation.portraitUp` and `[DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]`.

**Comment by jmagman** [comment]
> Have you attempted to reproduce it in release mode? For us, it hasn't been observed in debug mode, we stil could see the warning log though.
> 
> For what it's worth, the issue happened while rotating between `DeviceOrientation.portraitUp` and `[DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]`.

Did not reproduce for me in release mode between up and [left, right].  I'll rely on ya'll to validate the crash no longer happens on the iOS 16 APIs.

**Comment by jmagman** [comment]
This was reverted https://github.com/flutter/engine/pull/37165

**Comment by b055man** [comment]
@jmagman @danagbemava-nc Why this is not marked as a bug? Any plans for pulling this change into 3.3.x line?

**Comment by jmagman** [comment]
> Why this is not marked as a bug?

How do you want this marked as a bug?  It's definitely a bug.

> Any plans for pulling this change into 3.3.x line?

We generally reserve hot fixes for very high priority fixes, like that the app won't compile, or it crashes for a high percentage of users.

> Stability of the release will be the overarching goal, so only high impactful and critical cherrypicks will be allowed across Dart and Flutter. 

https://github.com/flutter/flutter/wiki/Flutter-Cherrypick-Process

**Comment by b055man** [comment]
> > Why this is not marked as a bug?
> 
> How do you want this marked as a bug? It's definitely a bug.

I can see it's referenced in the project here https://github.com/orgs/flutter/projects/23 and it's not marked as a bug there, opposed to some other issues - hence the question.

> > Any plans for pulling this change into 3.3.x line?
> 
> We generally reserve hot fixes for very high priority fixes, like that the app won't compile, or it crashes for a high percentage of users.

@jmagman So, in other words this means this particular one is not planned for any 3.3.x hotfix release and is rather expected in the next major version of Flutter? Have you got any rough ETA of it maybe (weeks? months?)?

**Comment by jmagman** [comment]
> > > Why this is not marked as a bug?
> > 
> > 
> > How do you want this marked as a bug? It's definitely a bug.
> 
> I can see it's referenced in the project here https://github.com/orgs/flutter/projects/23 and it's not marked as a bug there, opposed to some other issues - hence the question.

Ah. I wouldn't worry about that, it's not part of our issue PR process, it's more for organizational purposes.  🙂 

> > > Any plans for pulling this change into 3.3.x line?
> > 
> > 
> > We generally reserve hot fixes for very high priority fixes, like that the app won't compile, or it crashes for a high percentage of users.
> 
> @jmagman So, in other words this means this particular one is not planned for any 3.3.x hotfix release and is rather expected in the next major version of Flutter? Have you got any rough ETA of it maybe (weeks? months?)?

Exactly, should show up in the next stable release.  Right now it's only in the master channel, you can follow along with where it is by looking at the branch location in https://github.com/flutter/flutter/commit/6ea01e6aec25405b56667b2beb3966a789f9db44, see https://github.com/flutter/flutter/wiki/Where's-my-Commit%3F#finding-the-versions-that-contain-framework-commit-x.

**Comment by b055man** [comment]
> Exactly, should show up in the next stable release. Right now it's only in the master channel, you can follow along with where it is by looking at the branch location in [6ea01e6](https://github.com/flutter/flutter/commit/6ea01e6aec25405b56667b2beb3966a789f9db44), see https://github.com/flutter/flutter/wiki/Where's-my-Commit%3F#finding-the-versions-that-contain-framework-commit-x.

Cool, thank you!

**Comment by jmagman** [comment]
This is available in 3.6.0-0.0.pre, currently in the [beta channel](https://docs.flutter.dev/development/tools/sdk/releases?tab=macos).
